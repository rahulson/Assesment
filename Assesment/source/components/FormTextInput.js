import React from 'react'
import { View, TextInput, StyleSheet } from 'react-native'
import { widthPercentageToDP as wp, heightPercentageToDP as hp } from 'react-native-responsive-screen';
import { ColorConstants } from '../constants/ColorConstants'
import PropTypes from 'prop-types';

const FormTextInput = (props) => {

    const { containerStyle, inputContainerStyle, ...restProps } = props

    return (
        <View style={[styles.container, containerStyle]}>
            <TextInput
                style={[styles.inputContainer, inputContainerStyle]}
                {...restProps}
            />
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        marginBottom: '5%',
        width: '100%'
    },
    inputContainer: {
        height: hp(6),
        backgroundColor: ColorConstants.WHITE,
        borderRadius: hp(1),
        borderWidth: 1.0,
        borderColor: ColorConstants.LIGHT_GREY,
        width: '100%',
        paddingHorizontal: hp(1)
    }
})

FormTextInput.propTypes = {
    containerStyle: PropTypes.object,
    inputContainerStyle: PropTypes.object
};

FormTextInput.defaultProps = {
    containerStyle: {},
    inputContainerStyle: {}
};

export default FormTextInput