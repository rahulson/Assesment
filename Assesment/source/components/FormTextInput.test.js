import React from 'react';
import { render, fireEvent } from '@testing-library/react-native'
import FormTextInput from './FormTextInput';


test('should test placeholder and should call onChangeText when user is typing', () => {
    const onChangeTextEvent = jest.fn();
    const placeholderLabel = 'Enter Username';
    const { getByPlaceholderText, toJSON } = render(
        <FormTextInput placeholder={placeholderLabel} onChangeText={onChangeTextEvent} />,
    );
    const textInputInstance = getByPlaceholderText(placeholderLabel);
    expect(textInputInstance).toBeDefined();
    fireEvent(textInputInstance, 'onChangeText', 'unitesting');
    expect(onChangeTextEvent).toHaveBeenCalledWith('unitesting');


    expect(toJSON()).toMatchSnapshot();
});