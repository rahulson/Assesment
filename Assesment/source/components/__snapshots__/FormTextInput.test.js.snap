// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test placeholder and should call onChangeText when user is typing 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": "5%",
        "width": "100%",
      },
      Object {},
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={
      [MockFunction] {
        "calls": Array [
          Array [
            "unitesting",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    placeholder="Enter Username"
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderColor": "#c8c8c8",
          "borderRadius": 13.5,
          "borderWidth": 1,
          "height": 80,
          "paddingHorizontal": 13.5,
          "width": "100%",
        },
        Object {},
      ]
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
